### PR TITLE
Deploy workflow

### LINKED ISSUE #9: Docker dependency caching
Properly cache dependencies between Docker builds. There's a possible solution [here](https://github.com/rust-lang/cargo/issues/2644#issuecomment-335272535)

### PR DESCRIPTION
**Pull request by cjdenio**
Closes #9

### CODE REVIEW

File: scripts/deploy [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        -v /var/run/docker.sock:/var/run/docker.sock \
+        haas_api)
+    
+    HAAS_API_NEXT_IP=\$(docker inspect \$HAAS_API_NEXT | jq -r .[0].NetworkSettings.Networks.haas_admin.IPAddress)
+
+    echo "New container IP: \$HAAS_API_NEXT_IP"
+    
+    # Wait 5 seconds for the API to start up
+    echo -e "\nWaiting impatiently for HaaS to start up..."
+    sleep 5
```
Comment (jasonappah) [inline scripts/deploy:57]: Ngl I'm a little sus of this - ideally we'd block until we can confirm for sure the API is up, but that'd be a little more difficult in a bash script. How long does it take for the API to start up now?

File: scripts/deploy [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    echo -e "\nSwitching reverse proxy over to new deployment..."
+    # Update Caddy config
+    curl -fs -X PATCH -H "Content-Type: application/json" -d "{\"@id\": \"haas_upstream\", \"dial\": \"\$HAAS_API_NEXT_IP:3000\"}" http://localhost:2019/id/haas_upstream
+
+    echo -e "\nRenaming containers..."
+    docker rename haas_api haas_api_prev || true
+    docker rename haas_api_next haas_api
+
+    echo -e "\nWaiting to make sure all is well with the reverse proxy..."
+    sleep 5
```
Comment (jasonappah) [inline scripts/deploy:68]: Same as above - how long does it usually take for Caddy to settle after we make a config update?

**Review by jasonappah — COMMENTED** [review]
I have a couple questions but other than that everything looks good to me!

File: scripts/deploy (hunk 1 above)
Comment (cjdenio) [reply scripts/deploy:57]: It starts up instantly; the 5-second wait is just a safeguard

File: scripts/deploy (hunk 2 above)
Comment (cjdenio) [reply scripts/deploy:68]: Again, it's just a paranoid safeguard (Caddy config updates are instant too)

File: scripts/deploy (hunk 1 above)
Comment (jasonappah) [reply scripts/deploy:57]: Thanks!

File: scripts/deploy (hunk 2 above)
Comment (jasonappah) [reply scripts/deploy:68]: Got it, sounds good

**Review by jasonappah — APPROVED** [review]

**Pull request merged**